drivers/dma/dma_mcux_lpc.c

```diff
@@ -16,14 +16,16 @@
 #include <zephyr/logging/log.h>
 #include <zephyr/irq.h>
 #include <zephyr/sys/barrier.h>
+#include <zephyr/sys/util.h>
+#include <zephyr/sys/util_macro.h>
 
 #define DT_DRV_COMPAT nxp_lpc_dma
 
 LOG_MODULE_REGISTER(dma_mcux_lpc, CONFIG_DMA_LOG_LEVEL);
 
 struct dma_mcux_lpc_config {
 	DMA_Type *base;
-	uint32_t num_of_channels;
+	uint8_t num_of_channels;
```

Comment:
@decsny, this probably doesn't save anything to change to 8bit unless the compiler options are set to compress this structure. Otherwise it will align the following fields. An option that will likely work though would be to move the num_of_channels down to num_of_otrigs. You can probably also verify this by looking at the map file.

Reply:
taking this suggestion does in fact save a byte, pr updated